src/overlay/Floodgate.cpp

```diff
@@ -120,19 +124,32 @@ Floodgate::broadcast(StellarMessage const& msg, bool force)
         releaseAssert(peer.second->isAuthenticated());
         if (peersTold.insert(peer.second->toString()).second)
         {
-            mSendFromBroadcast.Mark();
-            std::weak_ptr<Peer> weak(
-                std::static_pointer_cast<Peer>(peer.second));
-            mApp.postOnMainThread(
-                [smsg, weak, log = !broadcasted]() {
-                    auto strong = weak.lock();
-                    if (strong)
-                    {
-                        strong->sendMessage(smsg, log);
-                    }
-                },
-                fmt::format(FMT_STRING("broadcast to {}"),
-                            peer.second->toString()));
+            if (msg.type() == TRANSACTION && peer.second->shouldFloodLazily())
+            {
+                mMessagesAdvertized.Mark();
+                TracyPlot("overlay.flood.advertized",
+                          static_cast<int64_t>(mMessagesAdvertized.count()));
+                releaseAssert(hash.has_value());
```

Comment:
> this assert should be moved to the top of the function as it's a bug to not pass down the hash when flooding a transaction: we want to require that callers always pass in a hash for a transaction (so that when we run tests etc we catch issues even if we have pull mode disabled).

Fixed

> Peer::sendMessage metrics should allow to see the effect of pull mode

I'm not sure what you mean.